src/backends/caffe/caffeinputconns.cc

```diff
@@ -278,26 +291,29 @@ namespace dd
     std::unique_ptr<db::Transaction> txn(db->NewTransaction());
     
     // Storing to db
-    Datum datum;
     int count = 0;
     const int kMaxKeyLength = 256;
     char key_cstr[kMaxKeyLength];
     
     for (int line_id = 0; line_id < (int)lfiles.size(); ++line_id) {
+      Datum datum;
```

Comment:
Is it necessary to create a new Datum for each file?

Reply:
Datum is the caffe inner representation of a sample before it is loaded up as a Blob object in  cpu/gpu memory.